tests/runtime/prelude/core/primitives/String.sk

```diff
@@ -35,8 +37,13 @@ native class .String uses Hashable, Show, Orderable {
         throwOutOfBounds()
       };
 
-      v = Array::fillBy(len, i -> this.unsafe_get(start + i));
-      static::fromChars(v)
+      // We don't want to use unsafe_get because it is O(n)
+      subString = mutable Vector[];
+      iter = this.getIter().drop(start);
+      for (_ in Range(0, len)) {
+        subString.push(iter.next().fromSome())
+      };
+      static::fromChars(freeze(subString.toArray()))
```

Comment:
You can use `StringIterator.substring(endIterator)`
https://github.com/skiplang/skip/blob/master/src/runtime/prelude/native/StringIterator.sk#L55

```js
iter = this.getIter().drop(start);
iter.substring(iter.clone().drop(end - start));
```

Reply:
@pikatchu you also want to do this change